src/ts/components/subtitleoverlay.ts

```diff
@@ -86,10 +101,96 @@ export class SubtitleOverlay extends Container<ContainerConfig> {
       }
     });
 
+    this.configureCea608Captions(player, uimanager);
     // Init
     subtitleClearHandler();
   }
 
+  configureCea608Captions(player: bitmovin.PlayerAPI, uimanager: UIInstanceManager): void {
+
+    let ratio = 1;
+    let generateCea608Ratio = () => {
+      let dummyText = 'aaaaaaaaaa';
+      let label = new SubtitleLabel({
+        // One letter label used to calculate the height wifth ratio of the font
+        // Works because we are using a monospace font for cea 608
+        // Using a longer string increases precision due to width being an integer
+        text: dummyText,
+      });
+      let domElement = label.getDomElement();
+      domElement.addClass(this.prefixCss(SubtitleOverlay.CLASS_CEA_608));
+      domElement.css({
+        'color': 'rgba(0, 0, 0, 0)',
+        'font-size': '30px',
+        'line-height': '30px',
+        'top': '0',
+        'left': '0',
+      });
+      this.addComponent(label);
+      this.updateComponents();
+      this.show();
+
+      let width = domElement.width() / dummyText.length;
+      let height = domElement.height();
+
+      this.removeComponent(label);
+      this.updateComponents();
+      if (!this.subtitleManager.hasCues) {
+        this.hide();
+      }
+
+      ratio = height / width;
+      this.cea608lineHeight = (new DOM(player.getFigure()).width()) * (SubtitleOverlay.CEA608_WIDTH / SubtitleOverlay.CEA608_NUM_COLUMN) * ratio * SubtitleOverlay.CEA608_FONT_SPACING;
+
+      if (this.isCEA608) {
+        for (let label of this.getComponents()) {
+          if (label instanceof SubtitleLabel) {
+            // Only element with left property are cea-608
+            let domElement = label.getDomElement();
+            domElement.css({
+              'font-size': `${SubtitleOverlay.CEA608_LINE_TO_FONT_SIZE * this.cea608lineHeight}px`,
+            });
+          }
+        }
+      }
+    };
+    player.addEventHandler(player.EVENT.ON_PLAYER_RESIZE, generateCea608Ratio);
+
+    player.addEventHandler(player.EVENT.ON_CUE_ENTER, (event: SubtitleCueEvent) => {
+      let labelToAdd = this.subtitleManager.getCue(event);
```

Comment:
Like written above, `getCue` (which should be `getCues`) returns an already added label, so you just have to apply CEA-608 styling here. No need to add it here, so no need for the `addLabel` function. Unless something speaks against it?